src/features/content-answers/ContentAnswersModal.tsx

```diff
@@ -48,15 +49,17 @@ const ContentAnswersModal = ({ api, currentUser, file, isOpen, onRequestClose }:
         if (lastQuestion && lastQuestion.prompt) {
             lastQuestion.answer = response.data.answer;
             lastQuestion.createdAt = response.data.created_at;
+            setQuestions([...updatedQuestions, lastQuestion]);
         }
-
-        setQuestions([...updatedQuestions, lastQuestion]);
     }, []);
 
-    const handleErrorCallback = useCallback((error: ElementsXhrError): void => {
+    const handleErrorCallback = useCallback((error: ElementsXhrError, updatedQuestions: QuestionType[]): void => {
         setIsLoading(false);
```

Comment:
why is `setIsLoading` called at the beginning of the function instead of the end of the logic or at the end of `handleOnAsk`?